src/wallet/output_foundry.c

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+
+#include "core/models/outputs/storage_deposit.h"
+#include "wallet/output_alias.h"
+#include "wallet/output_foundry.h"
+
+int wallet_foundry_output_mint_native_tokens(iota_wallet_t* w, address_t* alias_addr, bool state_ctrl_change,
+                                             uint32_t state_ctrl_index, address_t* govern_addr,
+                                             address_t* receiver_addr, uint256_t* max_supply, uint256_t* minted_tokens,
+                                             uint32_t serial_number, uint32_t foundry_counter,
+                                             res_send_message_t* msg_res) {
+  if (w == NULL || alias_addr == NULL || govern_addr == NULL || max_supply == NULL || minted_tokens == NULL ||
+      receiver_addr == NULL || msg_res == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  int ret = 0;
+  uint256_t* melted_tokens = NULL;
+  token_scheme_t* token_scheme = NULL;
+  utxo_outputs_list_t* outputs = NULL;
+  output_foundry_t* output_foundry = NULL;
+  output_basic_t* output_basic = NULL;
+  native_tokens_list_t* native_tokens = NULL;
+  cond_blk_list_t* unlock_conds = NULL;
+  unlock_cond_blk_t* unlock_addr = NULL;
+
+  // there is no melted native tokens
+  melted_tokens = uint256_from_str("0");
+  if (!melted_tokens) {
+    printf("[%s:%d] can not create melted tokens object\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // create token scheme
+  token_scheme = token_scheme_simple_new(minted_tokens, melted_tokens, max_supply);
+  if (!token_scheme) {
+    printf("[%s:%d] can not create token scheme object\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // create Foundry Output
+  output_foundry = output_foundry_new(alias_addr, 0, NULL, serial_number, token_scheme, NULL, 0, NULL, 0);
+  if (!output_foundry) {
+    printf("[%s:%d] can not create foundry output\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // calculate minimum storage deposit for Foundry Output and update output's amount
+  uint64_t min_storage_deposit = calc_minimum_output_deposit(&w->byte_cost, OUTPUT_FOUNDRY, output_foundry);
+  output_foundry->amount = min_storage_deposit;
+  printf("Storage deposit for foundry output: %" PRIu64 "\n", output_foundry->amount);
```

Comment:
I think it's for debugging, should comment out in most of case.
```suggestion
  // printf("Storage deposit for foundry output: %" PRIu64 "\n", output_foundry->amount);
```

Reply:
Debug outputs are commented out.